lambda/update/attorney_opt_out.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/validate"
+)
+
+type AttorneyOptOut struct {
+	AttorneyUID string
+}
+
+func (c AttorneyOptOut) Apply(lpa *shared.Lpa) []shared.FieldError {
+	attorney, ok := lpa.GetAttorney(c.AttorneyUID)
+	if !ok {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney not found"}}
+	}
+
+	if attorney.SignedAt != nil && !attorney.SignedAt.IsZero() {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney cannot opt out after signing"}}
+	}
+
+	attorney.Status = shared.AttorneyStatusRemoved
+	lpa.PutAttorney(attorney)
+
+	attorneysCount := len(lpa.ActiveAttorneys()) + len(lpa.ActiveTrustCorporations())
+	switch attorneysCount {
+	case 0, 1:
```

Comment:
I'm now questioning the logic here myself 😂 would having one remaining attorney invalidate the LPA in all cases or only if its the jointly make decisions listed below?

Reply:
do you have a link to where this logic came from? (I would've thought a single attorney is fine for "jointly and severally" if the other attorneys can be removed)

Reply:
That's the problem, I thought I could rely on the delete attorney MRLPA logic but as its pre-signing we're not focused on validity there. I've asked Abbi to add this topic to collab time for tomorrow to talk through the various scenarios. If you're happy to stick with this for now we could tweak in a follow up if it turns out the logic isn't quite right.